static/js/video-apis/youtube/YoutubePlayer.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Manages the state and interactions for an instance
+ * of the Youtube Iframe player.
+ */
+export default class YoutubePlayer {
+  /**
+   * @param {YT} YT the youtube iframe api
+   * @param {HTMLElement} el element to create the player for 
+   * @param {Function} onPlay runs when the play button is hit
+   */
+  constructor(YT, el, onPlay) {
+    this.YT = YT;
+    this.player = new YT.Player(el, {
+      events: {
+        onStateChange: event => this.onStateChange(event)
+      }
+    });
+    this.playerState = YT.PlayerState.UNSTARTED;
+    this.onPlay = onPlay;
+  }
+
+  /**
+   * @param {Object} param0 
+   * @param {PlayerState} param0.data
+   */
+  onStateChange({ data }) {
+    const isPlayingVideo = this.videoIsPlayingForState(data);
+    const wasAlreadyPlayingVideo = this.videoIsPlayingForState(this.playerState);
```

Comment:
would it be simpler to have a boolean value which keeps track of whether or not the video has started?

Reply:
we need to handle pausing the video, ending the video, cueing a new video

Reply:
oh, we want to send a new analytics event every time someone pauses and hits play again?